### PR TITLE
no dns service installed if seed has taint disabledDNS

### LINKED ISSUE #4: DNS service is installed for shoots without DNS domain
**What happened**:
Shoots that don't specify a DNS domain (`.spec.dns = nil`) or that are scheduled to a seed that is tainted with "DNS disabled" still get the DNS service.

**What you expected to happen**:
The DNS service shoot only be installed for shoots with a DNS domain.

**How to reproduce it (as minimally and precisely as possible)**:
Create a shoot with `.spec.dns=nil` and register the DNS extension (globally enabled). See it getting the DNS service pod.

- Gardener version (if relevant): v1.0.4
- Extension version: v1.3.0
- Kubernetes version (use `kubectl version`): doesn't matter
- Cloud provider or hardware configuration: doesn't matter

### PR DESCRIPTION
**Pull request by MartinWeindel**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #4 

**Special notes for your reviewer**:

**Release note**:

```improvement operator
no shoot DNS controller manager is installed if seed taint disabledDNS is set 
```

### CODE REVIEW

File: pkg/controller/actuator.go [hunk 1]
```diff
@@ -104,6 +107,12 @@ func (a *actuator) Reconcile(ctx context.Context, ex *extensionsv1alpha1.Extensi
 		return err
 	}
 
+	if gardencorev1beta1helper.TaintsHave(cluster.Seed.Spec.Taints, gardencorev1beta1.SeedTaintDisableDNS) ||
+		cluster.Shoot.Spec.DNS == nil {
+		a.logger.Info("no shoot dns service installed", "shoot", ex.Namespace)
```
Comment (rfranzke) [inline pkg/controller/actuator.go]: maybe add the reason here so that readers of the logs know why :)

Comment (rfranzke) [reply pkg/controller/actuator.go]: ping?

Comment (MartinWeindel) [reply pkg/controller/actuator.go]: already done, see commit below

Comment (rfranzke) [reply pkg/controller/actuator.go]: I was talking about the log message, not the comment in the code (which is also good to add, of course) :) - can you add it in the log msg as well + squash the commits?

**Pull request merged**